source/base/data_out_base.cc

```diff
@@ -2730,6 +2735,23 @@ namespace DataOutBase
 
     for (const auto &patch : patches)
       {
+        if (patch.reference_cell_type !=
+            ReferenceCell::Type::get_hypercube<dim>())
+          {
+            connectivity.resize(patch.data.n_cols());
+
+            for (unsigned int i = 0; i < patch.data.n_cols(); ++i)
+              connectivity[i] = i;
+
+            out.template write_high_order_cell<dim>(count++,
+                                                    first_vertex_of_patch,
+                                                    connectivity);
+
+            first_vertex_of_patch += patch.data.n_cols();
+
+            continue;
```

Comment:
I find `continue` statements inside `if` difficult to read. Just drop the `continue` here and make the rest of the body of the loop into the `else`. Or better yet, make the guarding `if` condition a positive and put the existing code into `if (cell = hypercube)` and put the new code at the bottom with `else`.

Reply:
If you do this right, the `if`...`else` would only contain setting up the data structures, and the call to `write_high_order_cell` and `first_vertex_of_patch += ...` would remain outside.